Events/Ch11.event

```diff
@@ -0,0 +1,720 @@
+        LoadReinforcements(RD3) // r =3
+    LABEL 0x23
+
+    NoFade
+    ENDA
+
+    SLu_Company:
+    RANDOMNUMBER 0x8       //Sets slot C to a number from 0 to 8? smilies says 9
+    //Now get remainder mod 4 (possibilities: 0, 1, 2, 3)
+    SMOV r1 0x4
+    SDIV slotsParam(r2, rC, r1) //r2 = rC/4
+    SMUL slotsParam(r1, r2)     //r1 = r2*4
+    SSUB slotsParam(r1, rC, r1)     //r1 -= rC, r1 = remainder
+    BNE 0x0 rC 0
+        LoadReinforcements(SweetLu_0)    //Remainder was 0
+        GOTO 0x3
+    LABEL 0x0
+    SMOV r2 1
+    BNE 0x1 rC r2
+        LoadReinforcements(SweetLu_1) //r = 1
+        GOTO 0x3
+    LABEL 0x1
+    SMOV r2 2
+    BNE 0x2 rC r2
+        LoadReinforcements(SweetLu_2) // r = 2
+        GOTO 0x3
+    LABEL 0x2
+        LoadReinforcements(SweetLu_3) // r =3
+    LABEL 0x3
+
+    RANDOMNUMBER 0x8       //Sets slot C to a number from 0 to 8? smilies says 9
+    //Now get remainder mod 4 (possibilities: 0, 1, 2, 3)
+    SMOV r1 0x4
+    SDIV slotsParam(r2, rC, r1) //r2 = rC/4
+    SMUL slotsParam(r1, r2)     //r1 = r2*4
+    SSUB slotsParam(r1, rC, r1)     //r1 -= rC, r1 = remainder
+    BNE 0x4 rC 0
+        LoadReinforcements(LT_a0)    //Remainder was 0
+        GOTO 0x7
+    LABEL 0x4
+    SMOV r2 1
+    BNE 0x5 rC r2
+        LoadReinforcements(LT_a1) //r = 1
+        GOTO 0x7
+    LABEL 0x5
+    SMOV r2 2
+    BNE 0x6 rC r2
+        LoadReinforcements(LT_a2) // r = 2
+        GOTO 0x7
+    LABEL 0x6
+        LoadReinforcements(LT_a3) // r =3
+    LABEL 0x7
+
+    RANDOMNUMBER 0x8       //Sets slot C to a number from 0 to 8? smilies says 9
+    //Now get remainder mod 4 (possibilities: 0, 1, 2, 3)
+    SMOV r1 0x4
+    SDIV slotsParam(r2, rC, r1) //r2 = rC/4
+    SMUL slotsParam(r1, r2)     //r1 = r2*4
+    SSUB slotsParam(r1, rC, r1)     //r1 -= rC, r1 = remainder
+    BNE 0x8 rC 0
+        LoadReinforcements(LT_b0)    //Remainder was 0
+        GOTO 0xB
+    LABEL 0x8
+    SMOV r2 1
+    BNE 0x9 rC r2
+        LoadReinforcements(LT_b1) //r = 1
+        GOTO 0xB
+    LABEL 0x9
+    SMOV r2 2
+    BNE 0xA rC r2
+        LoadReinforcements(LT_b2) // r = 2
+        GOTO 0xB
+    LABEL 0xA
+        LoadReinforcements(LT_b3) // r =3
+    LABEL 0xB
+
+    RANDOMNUMBER 0x8       //Sets slot C to a number from 0 to 8? smilies says 9
+    //Now get remainder mod 4 (possibilities: 0, 1, 2, 3)
+    SMOV r1 0x4
+    SDIV slotsParam(r2, rC, r1) //r2 = rC/4
+    SMUL slotsParam(r1, r2)     //r1 = r2*4
+    SSUB slotsParam(r1, rC, r1)     //r1 -= rC, r1 = remainder
+    BNE 0xC rC 0
+        LoadReinforcements(LT_c0)    //Remainder was 0
+        GOTO 0xF
+    LABEL 0xC
+    SMOV r2 1
+    BNE 0xD rC r2
+        LoadReinforcements(LT_c1) //r = 1
+        GOTO 0xF
+    LABEL 0xD
+    SMOV r2 2
+    BNE 0xE rC r2
+        LoadReinforcements(LT_c2) // r = 2
+        GOTO 0xF
+    LABEL 0xE
+        LoadReinforcements(LT_c3) // r =3
+    LABEL 0xF
+
+    RANDOMNUMBER 0x8       //Sets slot C to a number from 0 to 8? smilies says 9
```

Comment:
Tbh you use this pattern several times. You can probably make a subroutine and pass in the unit pointers to spawn in slots. 
Then just `CALL SpawnRandomUnitGroup` or smth.